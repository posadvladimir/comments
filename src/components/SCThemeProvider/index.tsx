
import { ThemeProvider } from 'styled-components';

import type { TChildrenProps } from 'src/ts/types/children';

import theme from 'src/styles/theme';

const SCThemeProvider: React.FC<TChildrenProps> = ({ children }) => (
    <ThemeProvider theme={theme}>
        {children}
    </ThemeProvider>
);

export default SCThemeProvider;