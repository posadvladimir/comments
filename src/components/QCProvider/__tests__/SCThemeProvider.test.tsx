
import { render, screen } from '@testing-library/react';

import { runSnapshotTest } from 'src/common/tests/snapshot';

import SCThemeProvider from '..';

describe('SCThemeProvider', () => {
    const childComponentTestText = 'Test Child';

    const component = (
        <SCThemeProvider>
            <div>{childComponentTestText}</div>
        </SCThemeProvider>
    );

    test('отображаются дочерние компоненты', () => {
        render(component);

        expect(screen.getByText(childComponentTestText)).toBeInTheDocument();
    });

    runSnapshotTest(component);
});