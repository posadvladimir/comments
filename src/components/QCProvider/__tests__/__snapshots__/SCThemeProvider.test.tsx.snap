// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SCThemeProvider соотвествует снепшоту 1`] = `
<div>
  Test Child
</div>
`;
